src/plugins/intel_npu/include/openvino/runtime/intel_npu/properties.hpp

```diff
@@ -0,0 +1,90 @@
+//
+// Copyright (C) 2022 Intel Corporation.
+// SPDX-License-Identifier: Apache 2.0
+//
+
+#pragma once
+
+#include <openvino/runtime/properties.hpp>
+
+namespace ov {
+namespace intel_npu {
+
+/**
+ * @brief [Only for NPU plugin]
+ * Type: uint64_t
+ * Read-only property to get size of already allocated NPU DDR memory (both for discrete/integrated NPU devices)
+ *
+ * Note: Queries driver both for discrete/integrated NPU devices
+ */
+static constexpr ov::Property<uint64_t, ov::PropertyMutability::RO> device_alloc_mem_size{"NPU_DEVICE_ALLOC_MEM_SIZE"};
+
+/**
+ * @brief [Only for NPU plugin]
+ * Type: uint64_t
+ * Read-only property to get size of available NPU DDR memory (both for discrete/integrated NPU devices)
+ *
+ * Note: Queries driver both for discrete/integrated NPU devices
+ */
+static constexpr ov::Property<uint64_t, ov::PropertyMutability::RO> device_total_mem_size{"NPU_DEVICE_TOTAL_MEM_SIZE"};
+
+/**
+ * @brief [Only for NPU plugin]
+ * Type: uint32_t
+ * Read-only property to get VPU driver version (for both discrete/integrated VPU devices)
+ */
+static constexpr ov::Property<uint32_t, ov::PropertyMutability::RO> driver_version{"NPU_DRIVER_VERSION"};
+
+/**
+ * @brief Read-only property to get the name of used backend
+ */
+static constexpr ov::Property<std::string, ov::PropertyMutability::RO> backend_name{"NPU_BACKEND_NAME"};
+
+/**
+ * @brief Defines the options corresponding to the legacy set of values.
+ */
+enum class LegacyPriority {
+    LOW = 0,     //!<  Low priority
+    MEDIUM = 1,  //!<  Medium priority
+    HIGH = 2     //!<  High priority
+};
+
+inline std::ostream& operator<<(std::ostream& os, const LegacyPriority& priority) {
```

Comment:
Is this header only considered as public API for plugin?
Add "OPENVINO_API" macro and hide implementation in `cpp` file.

Reply:
Have moved the properties.hpp to src/inference/include/openvino/runtime/intel_npu/properties.hpp to align with other plugins.  Also move the implementation to internal vpux_private_properties.hpp, not expose now. Please help to review again.